### ISSUE TITLE
Add "non-anonymous" mode using Tor's recent 0.2.9.8

### ISSUE THREAD

**Issue by GitHubPleaseDontBlockTor** [micahflee/onionshare#327]
It would be really nice to have a "non-anonymous" mode where the client looses anonymity but gains increased speeds and this using that new feature in Tor 0.2.9.8 : https://blog.torproject.org/blog/whats-new-tor-0298 To add that one should change torrc to 

       HiddenServiceNonAnonymousMode 1
       HiddenServiceSingleHopMode 1 

Of course, one should warn users before enabling this mode, perhaps even with a completely red gui.

This will also result in less load on the network, as there will only be 3 hops instead of 6.

(Side note: This is a repost, for some reason Github locked my other account where I made plenty of proposals including in the Tor Browser Launcher and onionshare website... Github please stop blocking tor users)

**Comment by micahflee** [comment]
(Sorry github blocked you.)

As soon as is feasible I want to completely remove support for the old way of creating onion services -- adding torrc settings using the control port -- and replace it with only ephemeral onion services. To make my life way easier, OnionShare depends on [stem](https://stem.torproject.org/) to talk to the control port.

I like this idea, but I think it will be blocked by at the very least stem supporting non-anonymous ephemeral onion services in their `create_ephemeral_hidden_service` function: https://stem.torproject.org/api/control.html#stem.control.Controller.create_ephemeral_hidden_service

It's awesome that this is a new tor feature, but it'll take a little bit of time for the ecosystem to catch up before we can add this to OnionShare. But I'll for it, once everything is ready.

**Comment by GitHubHatesTor** [comment]
lol, I've never expected someone using Tor in this hub...
If you're blocked(like your posts smells like "spammer"), all you have to do is send angry email to Github contact. And they'll unblock u.

I've locked third times with this account lol

Anyway, @micahflee, any chance to use "Github on ONION"? Because github is HOSTILE to privacy(they log each user's activity, all contributions are public, etc).

http://git.vola7ileiax4ueow.onion/

**Comment by micahflee** [comment]
That documentation for the `create_ephemeral_hidden_service` function actually says, "Changed in version 1.5.0: Added support for non-anonymous services." But it's not obvious how to use it. I emailed the tor-dev mailing list asking for help. But if this turns out to be simple, maybe it is possible to add this feature right away.

**Comment by micahflee** [comment]
Based on [this thread](https://lists.torproject.org/pipermail/tor-dev/2016-December/011773.html) on tor-dev, I'm gonna go ahead and keep this blocked for the foreseeable future, until there's better support for ephemeral non-anonymous onion services.

Basically, there's no way to create a non-anon onion service using ADD_ONION (essentially the `create_ephemeral_hidden_service` function in stem). If you want to use it, you need to change the Tor settings globally, which risks accidentally making some onion services non-anonymous that you don't intend to be.
